neqo-transport/src/packet/mod.rs

```diff
@@ -644,32 +645,45 @@ impl<'a> Public<'a> {
 
         // Generic long header.
         let version = Self::opt(decoder.decode_uint())?;
-        let dcid = ConnectionIdRef::from(Self::opt(decoder.decode_vec(1))?).into();
-        let scid = ConnectionIdRef::from(Self::opt(decoder.decode_vec(1))?).into();
+        let dcid = ConnectionIdRef::from(Self::opt(decoder.decode_vec(1))?);
+        let scid = ConnectionIdRef::from(Self::opt(decoder.decode_vec(1))?);
```

Comment:
This part should have a modest positive performance impact, because it avoids assignments on packets that end up being discarded.

I'm a little concerned that we're seeing a drop in performance in benchmarks though; there are no other changes that should alter performance.  Unless you count having three more bytes of transport parameters.

Reply:
> This part should have a modest positive performance impact, because it avoids assignments on packets that end up being discarded.

Not calling `into()` right away prevents a stack allocation only, right?

https://github.com/mozilla/neqo/blob/b01d9662cf6756880b2888dd95af663f372f2c10/neqo-transport/src/cid.rs#L88-L92

I want to make sure I am not missing something. I am fine either way.